content/blog/2019-12-compat.md

```diff
@@ -0,0 +1,250 @@
++++
+date = "2019-12-11"
+title = "Announcing Tokio-Compat"
+description = "December 11, 2019"
+menu = "blog"
+weight = 984
++++
+
+The [release][release-02] of Tokio 0.2 was the culmination of a great deal of
+hard work from numerous contributors, and has brought several significant
+improvements to Tokio. Using `std::future` and async/await makes writing async
+code using Tokio [much more ergonomic][await], and a new scheduler
+implementation makes Tokio 0.2's thread pool [as much as 10x faster][10x].
+However, updating existing Tokio 0.1 projects to use 0.2 and `std::future` poses
+some new challenges. Therefore, we're very excited to announce the release of
+the `tokio-compat` crate to help ease this transition.
+
+## Motivation
+
+Both the transition from `futures` 0.1 to `std::future` and the breaking changes
+to Tokio's APIs in 0.2 have made it [challenging] to update existing projects to
+benefit from the variety of improvements Tokio 0.2 offers. Additionally, these
+breaking changes make it difficult for projects to migrate incrementally:
+instead, they must do so all at once, requiring a lot of effort and sometimes
+putting other work on hold until the migration is complete. In order to enable
+incremental migration, we need a compatibility layer that allows us to use code
+written against both the legacy Tokio 0.1/`futures` 0.1 APIs _and_ the new Tokio
+0.2/`std::future` APIs in the same project.
+
+The `futures` crate's [`compat` module][futures-compat] provides
+interoperability between `futures` 0.1 and `std::future` _future types_, such as
+implementing `std::future::Future` for a type that implements the `futures` 0.1
+`Future` This is a foundational part of the compatibility story, but, on its
+own, it is insufficient to allow most projects to be incrementally updated. Most
+code using Tokio relies on the _runtime services_ that Tokio provides. These
+runtime services include the ability to spawn other tasks; the I/O driver, which
+allows tasks to be notified by the operating system's async I/O APIs, and the
+timer. Futures which rely on Tokio 0.1's runtime services will fail when run on
+the Tokio 0.2 runtime (even if they are converted to the `std::future::Future`
```

Comment:
> Futures which rely on Tokio 0.1's runtime services will fail when run on
the Tokio 0.2 runtime 

took me a bit to understand what you meant here, might be worth elaborating on a little more or revisiting the wording.

Reply:
@yaahc I tried to reword this in https://github.com/tokio-rs/website/pull/378/commits/d64b86c1081c2d292f10a9bdb4e08818aeacbd96, but I'm not sure if the new version is better — what do you think?

Reply:
I know I'm late to reply but this looks great.